instance/models/rabbitmq_server.py

```diff
@@ -0,0 +1,137 @@
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+"""
+Instance app models - RabbitMQ Server model and manager
+"""
+
+# Imports #####################################################################
+
+import logging
+import random
+from urllib.parse import urlparse
+
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.db import models, transaction
+from django_extensions.db.models import TimeStampedModel
+
+from .utils import ValidateModelMixin
+
+
+# Logging #####################################################################
+
+logger = logging.getLogger(__name__)
+
+
+# Constants ###################################################################
+
+DEFAULT_INSTANCE_RABBITMQ_PORT = 5671
+
+
+# Models ######################################################################
+
+class RabbitMQServerManager(models.Manager):
+    """
+    Custom manager for the RabbitMQServer model.
+    """
+
+    def _create_default(self):
+        """
+        Create the default RabbitMQ server configured in the Django settings, if any.
+        """
+        if settings.DEFAULT_RABBITMQ_API_URL:
+            api_url = urlparse(settings.DEFAULT_RABBITMQ_API_URL)
+            amqps_url = urlparse(settings.DEFAULT_INSTANCE_RABBITMQ_URL)
+
+            defaults = {
+                'api_url': api_url.geturl().replace(':'.join([api_url.username, api_url.password]), ''),
+                'admin_username': api_url.username,
+                'admin_password': api_url.password,
+                'instance_host': amqps_url.hostname,
+                'instance_port': amqps_url.port,
+            }
+
+            if not all(defaults.values()):
+                raise ImproperlyConfigured(
+                    "Found DEFAULT_RABBITMQ_API_URL, but it must contain basic auth credentials, and "
+                    "DEFAULT_INSTANCE_RABBITMQ_URL must contain both a hostname and port"
+                )
+            logger.info("Creating RabbitMQServer %s", defaults['instance_host'])
+
+            del defaults['api_url']
+            server, created = self.get_or_create(
+                api_url=settings.DEFAULT_RABBITMQ_API_URL,
+                defaults=defaults,
+            )
+            if not created and not all(getattr(server, name) == default for name, default in defaults.items()):
+                logger.warning(
+                    "RabbitMQServer for %s already exists, and its settings do not match "
+                    "the Django settings: " + ", ".join(["%s vs %s"] * len(defaults)),
+                    defaults['api_url'],
```

Comment:
Will fail because `defaults['api_url']` was [deleted above](https://github.com/open-craft/opencraft/pull/215/files#diff-26b5ed6e821ee2029046e9119ecdfe65R77).

Change to `settings.DEFAULT_RABBITMQ_API_URL`?